eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Deleg.hs

```diff
@@ -0,0 +1,255 @@
+      mRewardCoin == Just mempty 
+        ?! UnRegCertStakeKeyNonZeroAccountBalanceDELEG (UM.fromCompact <$> mRewardCoin)
+      maybeToStrictMaybe (UM.fromCompact <$> mDepositCoin) == sMayCoin
+        ?! UnRegCertStakeKeyDepositDoesNotMatchExpectedDepositDELEG (UM.fromCompact <$> mDepositCoin)
+      let umRDRemoved = Set.singleton stakeCred UM.⋪ UM.RewDepUView dsUnified
+          umSPoolRemoved = Set.singleton stakeCred UM.⋪ UM.SPoolUView umRDRemoved
+          newUMap = UM.PtrUView umSPoolRemoved UM.⋫ Set.singleton stakeCred
+      pure $
+        cState
+          { certDState =
+              certDState
+                { dsUnified = newUMap }
+          }
+    ConwayTxCertDeleg (ConwayDelegCert stakeCred delegatee) -> do
+      UM.member stakeCred (RewDepUView dsUnified)
+        ?! DelegCertStakeKeyNotRegisteredDELEG stakeCred
+      case delegatee of
+        DelegStake sPool -> do
+          let umSPoolRemoved = Set.singleton stakeCred UM.⋪ UM.SPoolUView dsUnified
+              newUMap = UM.SPoolUView umSPoolRemoved UM.⨃ Map.singleton stakeCred sPool
+          pure $
+            cState
+              { certDState =
+                  certDState
+                    { dsUnified = newUMap }
+              }
+        DelegVote dRep -> do
+          Set.member dRep vsDReps 
+            ?! DelegCertDRepNotRegisteredDELEG dRep
+          let umDRepRemoved = Set.singleton stakeCred UM.⋪ UM.DRepUView dsUnified
+              newUMap = UM.DRepUView umDRepRemoved UM.⨃ Map.singleton stakeCred dRep
+          pure $
+            cState
+              { certDState =
+                  certDState
+                    { dsUnified = newUMap }
+              }
+
+        DelegStakeVote sPool dRep -> do
+          Set.member dRep vsDReps 
+            ?! DelegCertStakeVoteDRepNotRegisteredDELEG dRep
+          let umSPoolRemoved = Set.singleton stakeCred UM.⋪ UM.SPoolUView dsUnified
+              umDRepRemoved = Set.singleton stakeCred UM.⋪ UM.DRepUView umSPoolRemoved
+              newSPool = UM.SPoolUView umDRepRemoved UM.⨃ Map.singleton stakeCred sPool
+              newUMap = UM.DRepUView newSPool UM.⨃ Map.singleton stakeCred dRep
+          pure $
+            cState
+              { certDState =
+                  certDState
+                    { dsUnified = newUMap }
+              }
+    ConwayTxCertDeleg (ConwayRegDelegCert stakeCred (DelegStakeVote sPool dRep) coin) -> do
+      coin == pd 
+        ?! RegDelegCertProvidedCoinNotEqualToExpectedPoolDepositDELEG coin
+      UM.notMember stakeCred (RewDepUView dsUnified) 
+        ?! RegDelegCertStakeKeyAlreadyRegisteredDELEG stakeCred
+      UM.notMember stakeCred (SPoolUView dsUnified) 
+        ?! RegDelegCertStakeKeyAlreadyRegisteredDELEG stakeCred
+      UM.notMember stakeCred (DRepUView dsUnified) 
+        ?! RegDelegCertDRepAlreadyRegisteredDELEG stakeCred
+      Set.member dRep vsDReps 
+        ?! RegDelegCertStakeVoteDRepNotRegisteredDELEG dRep
```

Comment:
We can get rid of this check - we don't care whether the DRep is registered or not.